bat-utils/lib/runtime-database.js

```diff
@@ -125,30 +125,61 @@ Database.prototype.get = function (collection, debug) {
   return result
 }
 
+// TODO: annotate this function and give it a more descriptive name
+Database.prototype.form = (index) => {
+  let result = ''
+
+  underscore.keys(index).forEach((key) => { result += '_' + key + '_' + index[key] })
+  return result.substr(1)
+}
+
+// TODO: annotate this function and give it a more descriptive name
+Database.prototype.gather = (entry) => {
+  const gather = (list) => {
+    const result = []
+
+    if (list) list.forEach((index) => { result.push(Database.prototype.form(index)) })
```

Comment:
good catch on my `this` binding failure. as @BrendanEich points out however it's best to not use arrow functions in prototype for this reason